op-chain-ops/cmd/celo-migrate/main.go

```diff
@@ -361,6 +355,24 @@ func runPreMigration(opts preMigrationOptions) ([]*rawdb.NumberHash, uint64, err
 		return nil
 	})
 	g.Go(func() error {
+		// Check if we should reset the non-ancient db
+		shouldReset := opts.resetNonAncients
+		if !shouldReset {
+			prevFullMigration, err := checkForPrevFullMigration(opts.newDBPath)
+			if err != nil {
+				return fmt.Errorf("failed to check for previous full migration: %w", err)
+			}
+			if prevFullMigration {
+				log.Info("Previous full migration detected", "process", "pre-migration")
+			}
+			shouldReset = prevFullMigration
+		}
+		if shouldReset {
+			log.Info("Resetting non-ancient db", "process", "pre-migration")
+			if err := cleanupNonAncientDb(opts.newDBPath); err != nil {
```

Comment:
Just to double check. The clean up is also erasing the mark, right?

Reply:
That's correct, the `cleanupNonAncientDb` function deletes all files except those in `/ancient`